force-app/main/default/classes/FlowControlLoopsCollections.cls

```diff
@@ -54,50 +60,78 @@ public with sharing class FlowControlLoopsCollections {
      * @param num2 The second number.
      * @return The maximum of the two numbers, or null if either number is null.
      */
-    public static Integer findMax(Integer num1, Integer num2) {
-        // Initialize largestNum as null
-        Integer largestNum = null;
-
-        return null; // Replace null with the variable you used to store the result
-    }
-
+        public static Integer findMax(Integer num1, Integer num2) {
+            //Check if either num1 or num2 is null
+            if (num1 == null || num2 == null){
+                return null;
+
+            }
+            Integer largestNum;
+            if (num1 > num2){
+                largestNum = num1;
+            } else {
+                largestNum = num2;
+            }
+            return largestNum;
```

Comment:
This logic works! But for problems like these, start exploring Math class methods. For instance, there's one called **[Math.max()](https://developer.salesforce.com/docs/atlas.en-us.apexref.meta/apexref/apex_methods_system_math.htm#apex_System_Math_max_3)** that would come in super handy here.